.github/workflows/ci.yml

```diff
@@ -51,3 +51,30 @@ jobs:
       - name: Generate FFI bindings
         run: just gen
       - run: cargo clippy --all-targets -- -D warnings
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: extractions/setup-just@v1
+      - name: Setup rust toolchain
+        run: rustup show
+      - uses: Swatinem/rust-cache@v2.2.0
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: "stable"
+      - name: Install FFI bindings
+        run: cargo install flutter_rust_bridge_codegen
+      - name: Start containers
+        run: |
+          docker-compose up -d
+          sleep 10 # We need to give docker a bit of time to startup
+
+      - name: Test containers are up
+        run: |
+          curl -d '{"address":"bcrt1qylgu6ffkp3p0m8tw8kp4tt2dmdh755f4r5dq7s", "amount":"0.1"}' -H "Content-Type: application/json" -X POST http://localhost:3000/faucet
+
+      - name: Generate FFI bindings
+        run: just gen
+      - name: Running cargo tests
+        run: cargo test
```

Comment:
The job is called `integration-test`, but here we are running all tests. Also, given our definition in the [CI strategy](https://github.com/get10101/10101/blob/main/docs/001-ci-strategy.adoc) I think our single and multi hop payment tests rather qualify as system tests as "They need a whole infrastructure to be set up to run."